tests/area/test_event_dispatcher.py

```diff
@@ -0,0 +1,253 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+from typing import Dict, Union
+from unittest.mock import MagicMock, Mock, call
+
+import pytest
+from d3a_interface.constants_limits import ConstSettings, GlobalConfig
+from d3a_interface.enums import SpotMarketTypeEnum
+from pendulum import DateTime, datetime
+from pendulum import duration
+
+from d3a.events.event_structures import MarketEvent, AreaEvent
+from d3a.models.area import Area
+from d3a.models.area.event_dispatcher import AreaDispatcher
+from d3a.models.market import Market
+from d3a.models.market.balancing import BalancingMarket
+from d3a.models.market.future import FutureMarkets
+from d3a.models.market.market_structures import AvailableMarketTypes
+from d3a.models.market.one_sided import OneSidedMarket
+from d3a.models.market.settlement import SettlementMarket
+from d3a.models.market.two_sided import TwoSidedMarket
+from d3a.models.strategy.area_agents.balancing_agent import BalancingAgent
+from d3a.models.strategy.area_agents.future_agent import FutureAgent
+from d3a.models.strategy.area_agents.inter_area_agent import InterAreaAgent
+from d3a.models.strategy.area_agents.one_sided_agent import OneSidedAgent
+from d3a.models.strategy.area_agents.settlement_agent import SettlementAgent
+from d3a.models.strategy.area_agents.two_sided_agent import TwoSidedAgent
+
+
+# pylint: disable=W0212
+
+
+@pytest.fixture(name="area_dispatcher")
+def area_dispatcher_fixture():
+    """Return fixture for AreaDispatcher object."""
+    area = Area("name")
+    area.children = [Area("child1"), Area("child2")]
+    area.parent = Area("parent")
+    return AreaDispatcher(area)
+
+
+@pytest.fixture(name="area_dispatcher_area_markets")
+def area_dispatcher_area_markets_fixture():
+    """Return fixture for AreaDispatcher object."""
+    config = Mock()
+    config.slot_length = duration(minutes=15)
+    config.tick_length = duration(seconds=15)
+    config.ticks_per_slot = 60
+    config.start_date = GlobalConfig.start_date
+    config.grid_fee_type = ConstSettings.IAASettings.GRID_FEE_TYPE
+    config.end_date = GlobalConfig.start_date + duration(days=1)
+
+    area = Area("name", config=config)
+    area.children = [Area("child1"), Area("child2")]
+    area.parent = Area("parent")
```

Comment:
I suggest creating an area fixture instead of the AreaDispatcher and reuse it here
you can build up the `AreaDispatcher` wherever you need later on, and I also don't have a problem if you decide to keep the `AreaDispatcher` and create a separate area fixture

Reply:
I'll do the latter actually.